contributors/design-proposals/vault-based-kms-provider.md

```diff
@@ -0,0 +1,302 @@
+	    	apiVersion: v1
+	    	cache-size: 100
+	    	config:
+	      	addr: https://localhost:8200
+	      	key-names:
+	            - kube-secret-enc-key
+	      	ca-cert:/var/run/kubernetes/ssl/vault.crt
+	      	client-cert:/var/run/kubernetes/ssl/vault-client-cert.pem
+	      	client-key:/var/run/kubernetes/ssl/vault-client-key.pem
+
+###### Vault token based authentication
+1. ``token`` : limited access vault token required by kubernetes API sever to
+authenticate itself while making requests to vault eg:
+8dad1053-4a4e-f359-2eab-d57968eb277f
+
+Here's a sample configuration file when using a Vault Token for authenticating
+the Kubernetes cluster as a client to Vault:
+
+	kind: EncryptionConfig
+	apiVersion: v1
+	resources:
+	  - resources:
+		- secrets
+		providers:
+	 	- kms:
+	    		kind: vault
+	    		apiVersion: v1
+	    		cache-size: 100
+	    		config:
+	      		addr: https://localhost:8200
+	      		key-names:
+	           		- kube-secret-enc-key
+	      		ca-cert:/var/run/kubernetes/ssl/vault.crt
+	      		token: 8dad1053-4a4e-f359-2eab-d57968eb277f
+
+###### Vault AppRole based authentication
+1. ``role-id`` : RoleID of the AppRole
+2. ``secret-id`` : secret Id only if associated with the appRole.
+
+Here's a sample configuration file with Vault AppRole
+	kind: EncryptionConfig
+	apiVersion: v1
+	resources:
+	  - resources:
+		- secrets
+		providers:
+	     - kms:
+	    	kind: vault
+	    	apiVersion: v1
+	    	cache-size: 100
+	    	config:
+	      	addr: https://localhost:8200
+	      	key-names:
+	            - kube-secret-enc-key
+	      	ca-cert: /var/run/kubernetes/ssl/vault.crt
+	      	role-id: db02de05-fa39-4855-059b-67221c5c2f63
+
+## Key Generation and rotation
+The KEK is generated in Vault and rotated using direct API call or CLI to Vault
+itself. The Key never leaves the vault.
+
+Note that when a key is rotated, Vault does not allow to choose a different
+encryption algorithm or key size. If a key for different encryption algorithm or
+a different key size is desired, new key needs to be generated in Vault and the
+corresponding key name be added in the configuration. Subsequent encryption will
+be done using the first key in the list. Decryption can happen using any of the
+keys in the list based on the prefix to the encrypted DEK.
+
+## Backward compatibility
+1. Unencrypted secrets and secrets encrypted using other non-KMS providers will
+continue to be readable upon adding vault as a new KMS provider.
+2. If a Vault KMS is added as first provider, the secrets created or modified
+thereafter will be encrypted by vault provider.
+
+## Performance
+1. KMS provider framework uses LRU cache to minimize the requests to KMS for
+encryption and decryption of DEKs.
+2. Note that there will be a request to KMS for every cache miss causing a
+performance impact. Hence, depending on the cache size, there will be a
+performance impact.
+3. Response time.
+    4. will depend on choice of encryption algorithm and strength.
```

Comment:
Still not fixed. If you look at the file preview, the bullet points are inline and not nested as you wanted.